### ISSUE TITLE
[BUG] Application window doesn't show

### ISSUE THREAD

**Issue by Exxar** [Openarl/PathOfBuilding#30]
This happens on my 4-year old Asus laptop with Windows 7. After installation the program works normally for some time, but then, without a pattern I could detect, the application window simply doesn't show up after the initial console window is done loading stuff. The only solution to this problem I have been able to find is to reinstall.

I thought it had something to do with me having an external monitor hooked up to my laptop as that sometimes produces minor issues with games, but this week I've had the monitor hooked up to a different computer and the issue appeared despite it, with only the laptop screen as the sole display.

**Comment by Openarl** [comment]
So the window doesn't show up, but does it appear in the taskbar? If so, it's possible it's appearing offscreen. Try moving the window using Windows+Shift+Left/Right.

The fact that it persists until you reinstall suggests that the program's saved window position is being corrupted somehow. I'll need to see the contents of "Launch.cfg". In a standalone installation it will be in the install folder; with a normal install you'll find it in "%ProgramData%\Path of Building". It's a text file, so you can just open it using Notepad and paste the contents here. Once you've done that, try deleting everything in the file (but don't delete the file itself) and see if that fixes the issue.

**Comment by Exxar** [comment]
Yes that's exactly what's happening, the window doesn't show up but the application sits in the taskbar.

I'll try it the next time it happens.

**Comment by Exxar** [comment]
Winkey + left/right rescued it from offscreen. Forgot to check Launch.cfg before doing it, but I'll do it the next time it happens to confirm your hypothesis.

**Comment by Openarl** [comment]
Ok all good. We're making progress at least :).

**Comment by Exxar** [comment]
So it happened again. Launch.cfg:

set con_log "0"
set r_compress "0"
set r_screenshotFormat "jpg"
set con_fontSize "14"
set vid_mode "8"
set vid_resizable "3"
set vid_display "-1"
set vid_fullscreen "0"
set vid_last "1280,720,-32000,-32000,0"

Closed the program, opened it again, and closed:

set con_log "0"
set r_compress "0"
set r_screenshotFormat "jpg"
set con_fontSize "14"
set vid_mode "8"
set vid_resizable "3"
set vid_display "-1"
set vid_fullscreen "0"
set vid_last "0,0,-32000,-32000,0"

Deleted contents of Launch.cfg, relauched the program, the window is there.

**Comment by Openarl** [comment]
Ok, so it's not being moved a little off screen, it's going wayyyy off. Hmm. I've poked around in the code that works with the window positioning and I can't for the life of me figure out that could happen. The saved position is only changed when Windows tells the program that the main window has been moved to the new position (via WM_MOVE), so it should never be invalid. I'll need to investigate more.
